src/Microsoft.Extensions.Configuration.UserSecrets/PathHelper.cs

```diff
@@ -3,13 +3,31 @@
 
 using System;
 using System.IO;
+using Microsoft.Extensions.FileProviders;
+using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Extensions.Configuration.UserSecrets
 {
     public class PathHelper
     {
         internal const string Secrets_File_Name = "secrets.json";
+        internal const string Config_File_Name = "project.json";
+
+        public static string GetSecretsPath(IFileProvider provider)
+        {
+            var fileInfo = provider.GetFileInfo(Config_File_Name);
+            if (fileInfo.PhysicalPath == null)
+            {
+                throw new InvalidOperationException(
+                    string.Format(Resources.Error_Missing_Project_Json, provider.GetFileInfo("/").PhysicalPath));
+            }
+
+            using (var stream = new StreamReader(fileInfo.CreateReadStream()))
+            {
+                return InternalGetSecretsPath(stream, fileInfo.Name);
+            }
+        }
 
         public static string GetSecretsPath(string projectPath)
```

Comment:
Can't this overload simply create a `PhysicalFileProvider` with the provided path and delegate to the `IFileProvider` overload?